editor/plugins/visual_shader_editor_plugin.cpp

```diff
@@ -2626,18 +2651,24 @@ void VisualShaderEditor::_update_graph() {
 	graph->set_connection_lines_curvature(graph_lines_curvature);
 }
 
-VisualShader::Type VisualShaderEditor::get_current_shader_type() const {
-	VisualShader::Type type;
-	if (mode & MODE_FLAGS_PARTICLES) {
-		type = VisualShader::Type(edit_type->get_selected() + 3 + (custom_mode_enabled ? 3 : 0));
-	} else if (mode & MODE_FLAGS_SKY) {
-		type = VisualShader::Type(edit_type->get_selected() + 8);
-	} else if (mode & MODE_FLAGS_FOG) {
-		type = VisualShader::Type(edit_type->get_selected() + 9);
-	} else {
-		type = VisualShader::Type(edit_type->get_selected());
+void VisualShaderEditor::_restore_editor_state() {
+	String id_string = visual_shader->get_path();
+	const ResourceUID::ID uid = ResourceLoader::get_resource_uid(id_string);
+	if (uid != ResourceUID::INVALID_ID) {
+		id_string = ResourceUID::get_singleton()->id_to_text(uid);
 	}
-	return type;
+
+	const int type = get_current_shader_type();
+	// const Vector2 saved_scroll_offset = EditorSettings::get_singleton()->get_project_metadata("visual_shader", uid_text + ":type" + itos(type) + ":offset", Vector2());
+	const Vector2 saved_scroll_offset = vs_editor_cache->get_value("visual_shader", id_string + ":type" + itos(type) + ":offset", Vector2());
+	// const real_t saved_zoom = EditorSettings::get_singleton()->get_project_metadata("visual_shader", uid_text + ":type" + itos(type) + ":zoom", 1.0);
+	const real_t saved_zoom = vs_editor_cache->get_value("visual_shader", id_string + ":type" + itos(type) + ":zoom", 1.0);
+
+	// Setting the scroll offset needs to be double-deferred because it requires min/max scroll offset to be final (as it gets clamped).
```

Comment:
Does it really make difference? "Double deferred" is not a thing, the calls are executed within the same flush (although something can be called in between).

Reply:
I agree, "double deferred" is bad wording, but the rest of the comment is still valid. Calling this deferred so that it's called at the end of the flush is necessary in this case. Adjusted the comment accordingly.